tools/legacy/sample_encode/src/pipeline_encode.cpp

```diff
@@ -2376,6 +2376,12 @@ mfxStatus CEncodingPipeline::EncodeOneFrame(const ExtendedSurface &In, sTask *&p
             sts = MFX_ERR_NONE; // ignore warnings if output is available
             if (m_nPerfOpt)
                 m_nEncSurfIdx++;
+
+            if (m_bAPI2XInternalMem)
```

Comment:
It seems checking "In.pSurface!=nullptr" is needed here (in case of flashing input surface maybe empty).

Reply:
Yes, it might be so. This PR was just to showcase how to fix #18, I didn't validate it, so it opened as draft.